src/components/TodoContainer/TodoContainer.jsx

```diff
@@ -0,0 +1,266 @@
+import React, { useState, useEffect } from "react";
+import AddTodoForm from "../AddTodo/AddTodoForm";
+import TodoList from "../TodoList/TodoList";
+import { Link } from "react-router-dom";
+
+const useDebounce = (value, delay) => {
+  const [debouncedValue, setDebouncedValue] = useState(value);
+
+  useEffect(() => {
+    const handler = setTimeout(() => {
+      setDebouncedValue(value);
+    }, delay);
+
+    return () => {
+      clearTimeout(handler);
+    };
+  }, [value, delay]);
+
+  return debouncedValue;
+};
+
+function TodoContainer() {
+  const [todoList, setTodoList] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [searchValue, setSearchValue] = useState("");
+  const [isAscending, setIsAscending] = useState(true);
+  const debouncedSearchValue = useDebounce(searchValue, 500);
+
+  const url = `https://api.airtable.com/v0/${
```

Comment:
Feels like there is opportunity to share code between this component and `CompletedTodoContainer`, especially around working with the API. Spend some time thinking about how you can abstract some of this logic out of both components and share functionality.